### DISCUSSION TITLE
Prevent query from returning cached data when using `skipToken`

### DISCUSSION THREAD

**Discussion by JosNun** [reduxjs/redux-toolkit#1492] [Q&A] [answered]
Is there a way to prevent a query from returning cached data when I'm passing skipToken as the arg? In my case, between selected items, a parameter changes from `a` to `undefined` to `b`, and while `b` is still fetching, the `data` field returned from the query is the value of `a`.

Ie, in the following example, if routeParams.code is A, then code changes to undefined, then code changes to B, while the data for B is fetching, data will contain the data for A.
```ts
const { data } =
    useGetItemByCodeQuery(
        routeParams.code
            ? {
                  code: routeParams.code,
              }
            : skipToken
    );
    
console.log(routeParams.code, data);
// a, {code: a}
// undefined, {code: a}
// b, {code: a}
// b, {code: b}
```

A potential workaround is utilizing the `isUnitialized` value returned from a query, but in my particular case, I have multiple queries dependent on the first, so there'll be a handful of `isInitialized` variables that'll need to be juggled so as not to show data for a different item.

**Comment by phryneas** [comment, marked as answer]
Not at the moment. You can only do something like
```js
const arg = routeParams.code ? { code: routeParams.code }  : skipToken
const { data } = useGetItemByCodeQuery(arg);
const myData = arg == skipToken ? undefined : data
```

**Reply by JosNun** [reply]
Thanks! I ended up doing something similar, it just felt a little unergonomic, but I guess this isn't that common of a use case. Thanks again!
